src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts

```diff
@@ -0,0 +1,1091 @@
+			const activeEditor = this.editorService.activeEditor;
+			if (!activeEditor && !hasBackups) {
+				const openWithReadme = this.configurationService.getValue(configurationKey) === 'readme';
+				if (openWithReadme) {
+					let readmes = await Promise.all(this.contextService.getWorkspace().folders.map(async folder => {
+						const folderUri = folder.uri;
+						try {
+							const folder = await this.fileService.resolve(folderUri);
+							const files = folder.children ? folder.children.map(child => child.name) : [];
+							const file = arrays.find(files.sort(), file => strings.startsWith(file.toLowerCase(), 'readme'));
+							if (file) {
+								return joinPath(folderUri, file);
+							}
+						} catch (err) {
+							onUnexpectedError(err);
+						}
+						return undefined;
+					}));
+					arrays.coalesceInPlace(readmes);
+					if (!this.editorService.activeEditor) {
+						if (readmes.length) {
+							const isMarkDown = (readme: URI) => strings.endsWith(readme.path.toLowerCase(), '.md');
+							await Promise.all([
+								this.commandService.executeCommand('markdown.showPreview', null, readmes.filter(isMarkDown), { locked: true }),
+								this.editorService.openEditors(readmes.filter(readme => !isMarkDown(readme))
+									.map(readme => ({ resource: readme }))),
+							]);
+						} else {
+							await this.instantiationService.createInstance(WelcomePage).openEditor();
+						}
+					}
+				} else {
+					await this.instantiationService.createInstance(WelcomePage).openEditor();
+				}
+			}
+		}
+	}
+}
+
+
+function isWelcomePageEnabled(configurationService: IConfigurationService, contextService: IWorkspaceContextService) {
+	const startupEditor = configurationService.inspect(configurationKey);
+	if (!startupEditor.userValue && !startupEditor.workspaceValue) {
+		const welcomeEnabled = configurationService.inspect(oldConfigurationKey);
+		if (welcomeEnabled.value !== undefined && welcomeEnabled.value !== null) {
+			return welcomeEnabled.value;
+		}
+	}
+	return startupEditor.value === 'welcomePage' || startupEditor.value === 'readme' || startupEditor.value === 'welcomePageInEmptyWorkbench' && contextService.getWorkbenchState() === WorkbenchState.EMPTY;
+}
+
+export class WelcomePageAction extends Action {
+
+	public static readonly ID = 'workbench.action.showWelcomePage';
+	public static readonly LABEL = localize('welcomePage', "Welcome");
+
+	constructor(
+		id: string,
+		label: string,
+		@IInstantiationService private readonly instantiationService: IInstantiationService
+	) {
+		super(id, label);
+	}
+
+	public run(): Promise<void> {
+		return this.instantiationService.createInstance(WelcomePage)
+			.openEditor()
+			.then(() => undefined);
+	}
+}
+
+interface ExtensionSuggestion {
+	name: string;
+	title?: string;
+	description?: string;
+	id: string;
+	isKeymap?: boolean;
+	isCommand?: boolean;
+	isExtensionPack?: boolean;
+	icon?: string;
+	link?: string;
+	extensionPackExtensions?: any[];
```

Comment:
nit: But to make this easier to use with typing I'd suggest creating an interface something like this 

```
interface ExtensionPackExtension {
name: string;
link: string;
icon: string; 
}
``` 

Although it looks like you currently aren't using the links - is that intended? You create the anchor tag but it doesn't link anywhere.